common/state_machine.hpp

```diff
@@ -202,6 +204,7 @@ class StateMachineRunner : virtual public EventPoster<EventType> {
 
 		if (!to_run.empty()) {
 			for (auto &state : to_run) {
+				log::Trace("Entering state " + common::BestAvailableTypeName(*state));
```

Comment:
Thanks!

However, why not just give the states a named string?

Reply:
Yeah, I did consider this too. The reason I opted for this method is because the states tend to be copy-pasted a lot, and you'd have to make sure that every copy-paste was followed by a change to this string. Very easy to miss. I also couldn't find any way to static-assert that two classes don't have the same strings, so I chose this method simply because it's less error prone.

But honestly, both methods suck, and I implemented this quite late in the process, because I was trying to avoid it. I just really needed to see the flow.

Very open to suggestions here, I just considered this the "least evil" method.